runtime/src/bank/builtins/core_bpf_migration/source_bpf_upgradeable.rs

```diff
@@ -0,0 +1,365 @@
+use {
+    super::error::CoreBpfMigrationError,
+    crate::bank::Bank,
+    solana_sdk::{
+        account::{AccountSharedData, ReadableAccount},
+        bpf_loader_upgradeable::{
+            get_program_data_address, UpgradeableLoaderState, ID as BPF_LOADER_UPGRADEABLE_ID,
+        },
+        pubkey::Pubkey,
+    },
+};
+
+/// Used to validate a source BPF Upgradeable program's account and data
+/// account before migrating a built-in program to Core BPF.
```

Comment:
```suggestion
/// The account details of an Upgradeable BPF program
/// slated to replace a built-in program.
```
The struct itself isn't used for validation. (I actually have the same comment on `TargetProgramBuiltin`. Will PR some comment change recommendations.)